src/App.tsx

```diff
@@ -1,9 +1,53 @@
-import React from 'react';
+import React, { useState} from 'react';
+import { TodoList } from './components/TodoList/TodoList';
+import {Todo} from "./components/todo.typedefs";
+import {AddForm} from "./components/AddForm/AddForm";
+
+const ShiftByType = {
+  'up': -1,
+  'down': 1,
+}
 
 function App() {
+  const [todos, setTodos] = useState<Todo[]>([]);
+
+  const addTodo = (place: Todo[]) => (title: string) => {
+    place.push({
+      id: Math.random(),
+      title,
+      items: [],
+    });
+
+    setTodos((todos) => [...todos]);
+  };
+
+  const move = (place: Todo[], type: keyof typeof ShiftByType) => (index: number) => {
```

Comment:
add enum MoveDirection

```suggestion
  const move = (place: Todo[], type: MoveDirection) => (index: number) => {
```